pkg/provisioner/core/provisioner.go

```diff
@@ -109,16 +109,29 @@ func mapAvailabilityDomainToFailureDomain(AD string) string {
 	return parts[len(parts)-1]
 }
 
+func resolveFSType(options controller.VolumeOptions) string {
+	return getFsTypeWithDefault(options.Parameters["fsType"])
+}
+
+func getFsTypeWithDefault(fsType string) string {
+	fs := "ext4"
+	if fsType != "" {
+		fs = fsType
+	}
+	return fs
+}
+
 // Provision creates a storage asset and returns a PV object representing it.
 func (p *OCIProvisioner) Provision(options controller.VolumeOptions) (*v1.PersistentVolume, error) {
 	availabilityDomainName, availabilityDomain, err := p.chooseAvailabilityDomain(options.PVC)
 	if err != nil {
 		return nil, err
 	}
 
-	provisioner, ok := p.storageClassProvisioners[*options.PVC.Spec.StorageClassName]
+	fsType := resolveFSType(options)
+	provisioner, ok := p.storageClassProvisioners[fsType]
```

Comment:
I'm still not overly happy with the overloading of fsType here. For one thing, it means creating multiple copies of the block storage provisioner in `p.storageClassProvisioners` unnecessarily. More importantly, it's not really what I'd expect as a user as throughout Kubernetes `FSType` is used to denote the filesystem type rather than the storage backend:

```go
	// Filesystem type to mount.
	// Must be a filesystem type supported by the host operating system.
	// Ex. "ext4", "xfs", "ntfs". The default filesystem depends on FlexVolume script.
	// +optional
	FSType string
```
[Source](https://github.com/kubernetes/kubernetes/blob/60cd056cb443c23b53240d274df4d5abc85c7aaf/pkg/apis/core/types.go#L746)

I suggest we do something akin to what [hyperkube does with symlinks](https://github.com/kubernetes/kubernetes/blob/master/cluster/images/hyperkube/Dockerfile#L23) in the flexvolume driver DaemonSet and then dispatch on `provisioner` in the StorageClass (i.e. `oracle/oci-block-storage` / `oracle/oci-filesystem-storage`).